### PR TITLE
Mobile: Update gradle, biometric error messages and fix modal issue

### PR DESCRIPTION
**Pull request by cvarley100**
# Description

- Adds more verbose Biometric error messages
- Fixes biometric modal issue

Fixes # (issue)

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Updates gradle

# How Has This Been Tested?

Tested on device

# Checklist:

_Please delete items that are not relevant._

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] For changes to `mobile` that include native code (including React Native modules): I have verified that both iOS and Android successfully build in both `Debug` and `Release` modes

### CODE REVIEW

**Comment by laumair** [comment]
Why are we removing `src/mobile/android/app/src/main/assets/index.android.bundle`?

**Comment by cvarley100** [comment]
It shouldn't be committed in the first place. We need to add it to gitignore.

**Comment by cvarley100** [comment]
> Why are we removing `src/mobile/android/app/src/main/assets/index.android.bundle`?

It shouldn't be committed in the first place. We need to add it to gitignore.

**Comment by rajivshah3** [comment]
I added `index.android.bundle` to the gitignore in #968. It might cause a merge conflict here, though

**Comment by laumair** [comment]
@cvarley100 Merge conflicts.

**Review by rajivshah3 — APPROVED** [review]

File: src/mobile/src/ui/components/EnterPassword.js [hunk 1]
```diff
@@ -103,7 +105,9 @@ class EnterPassword extends Component {
     }
 
     hideModal() {
-        this.props.toggleModalActivity();
+        if (this.props.isModalActive) {
+            this.props.toggleModalActivity();
```
Comment (laumair) [inline src/mobile/src/ui/components/EnterPassword.js]: What happens if `toggleModalActivity` is dispatched when modal is already inactive?

Comment (cvarley100) [reply src/mobile/src/ui/components/EnterPassword.js]: Well this function is just for hiding the modal.

Comment (laumair) [reply src/mobile/src/ui/components/EnterPassword.js]: Oh didn't realise it was for just hiding modal as it also [accepts](https://github.com/iotaledger/trinity-wallet/blob/develop/src/shared/actions/ui.js#L217) `modalProps` and `modalContent`. 

Perhaps the condition (if modal is active) is moved to the redux action itself. 

```
const hideModal = () => {
    return (dispatch, getState) => {
        if (getState().ui.isModalActive) { dispatch(toggleModalActivity()); };
    };
};
```

That way we won't need to include `isModalActive` prop in every component where we need to close the modal.

Comment (cvarley100) [reply src/mobile/src/ui/components/EnterPassword.js]: This was more as a failsafe. I don't think they are necessary however - I will remove them.

Comment (cvarley100) [reply src/mobile/src/ui/components/EnterPassword.js]: FYI, `toggleModalActivity` has two purposes - opening the right modal with the right props. But without props it simply closes an open modal.

**Pull request merged**